projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
+
+```typescript
+const Store = signalStore(
+  withEntities({ entity: type<Todo>(), collection: 'todo' }),
+  withEntities({
+    entity: type<User>(),
+    collection: 'user',
+  }),
+  withEntities({ entity: type<Category>(), collection: 'category' })
+);
+```
+
+Try to avoid multiple entity types in one store. It is better to have multiple stores, each with a single entity type.
+
+## Extending and Integrating
+
+You will usually want to persist your entities to a backend. Therefore, you must additionally implement `withMethods` and add the necessary methods.
+
+```typescript
+const TodoStore = signalStore(
+  withEntities<Todo>(),
+  withMethods((store) => {
+    const todoService = inject(TodoService);
+
+    return {
+      async load() {
+        const todos = await todoService.findAll();
+        patchState(store, removeAllEntities());
+        patchState(store, setEntities(todos));
+      },
+
+      async add(name: string) {
+        const todo = await todoService.add(name);
+        patchState(store, addEntity(todo));
+      },
+
+      async remove(id: number) {
+        await todoService.remove(id);
+        patchState(store, removeEntity(id));
+      },
+
+      async setFinished(id: number) {
+        await todoService.setFinished(id);
+        patchState(store, updateEntity({ id, changes: { finished: true } }));
+      },
+      async setUnfinished(id: number) {
+        await todoService.setUnfinished(id);
+        patchState(store, updateEntity({ id, changes: { finished: false } }));
+      },
+    };
+  }),
+  withHooks({ onInit: (store) => store.load() })
+);
+```
+
+If your API follows the same pattern, you can even create your own `with*` function:
+
+```typescript
+interface EntityService<Entity> {
+  findAll(): Promise<Entity[]>;
+
+  add(name: string): Promise<Entity>;
+
+  remove(id: number): Promise<void>;
+}
+
+function withPersistedEntities<Entity extends { id: number }>(EntityService: ProviderToken<EntityService<Entity>>) {
+  return signalStoreFeature(
+    withEntities<Entity>(),
+    withMethods((store) => {
+      const entityService = inject(EntityService);
+
+      return {
+        async load() {
+          const entities = await entityService.findAll();
+          patchState(store, removeAllEntities());
+          patchState(store, setEntities(entities));
+        },
+
+        async add(name: string) {
+          const entity = await entityService.add(name);
+          patchState(store, addEntity(entity));
+        },
+
+        async remove(id: number) {
+          await entityService.remove(id);
+          patchState(store, removeEntity(id));
+        },
+      };
+    })
+  );
+}
+
+const TodoStore = signalStore(withPersistedEntities<Todo>(TodoService), withHooks({ onInit: (store) => store.load() }));
+```
+
+The `withPersistedEntities` uses internally `withEntities` and adds the communication with the API via requesting an `EntityService`.
```

Comment:
```suggestion
The `withPersistedEntities` uses internally `withEntities` and adds the communication with the API through requesting an `EntityService`.
```

Reply:
Thanks @brandonroberts. Always good to have a native speaker around. Grammarly didn't detect everything.